src/transaction.ts

```diff
@@ -73,25 +74,41 @@ export interface KeyRange {
 }
 
 export interface ReadRequest extends RequestOptions {
-  keySet?: s.KeySet;
+  session?: string | null;
+  transaction?: TransactionSelector | null;
+  table?: string | null;
+  index?: string | null;
+  columns?: string[] | null;
   keys?: string[];
-  ranges?: KeyRange[];
-  index?: string;
-  columns?: string[];
-  limit?: number;
-  resumeToken?: ResumeToken;
-  partitionToken?: Uint8Array | string;
+  ranges?: spannerClient.spanner.v1.IKeyRange[];
+  keySet?: spannerClient.spanner.v1.IKeySet | null;
+  limit?: number | Long | null;
+  resumeToken?: Uint8Array | null;
+  partitionToken?: Uint8Array | null;
+}
+
+export interface TransactionSelector {
+  singleUse?: spannerClient.spanner.v1.ITransactionOptions | null;
+  id?: Uint8Array | string | null;
+  begin?: spannerClient.spanner.v1.ITransactionOptions | null;
 }
 
 export interface BatchUpdateError extends ServiceError {
   rowCounts: number[];
 }
 
+export type CommitRequest = spannerClient.spanner.v1.ICommitRequest;
+
 export type BatchUpdatePromise = Promise<[number[], s.ExecuteBatchDmlResponse]>;
-export type BeginPromise = Promise<[s.Transaction]>;
-export type CommitPromise = Promise<[s.CommitResponse]>;
+export type BeginPromise = Promise<[spannerClient.spanner.v1.ITransaction]>;
+export interface BeginTransactionCallback {
+  (err: Error | null, transaction: spannerClient.spanner.v1.ITransaction): void;
+}
+export type CommitPromise = Promise<[spannerClient.spanner.v1.CommitResponse]>;
 export type ReadPromise = Promise<[Rows]>;
-export type RunPromise = Promise<[Rows, s.ResultSetStats]>;
+export type RunPromise = Promise<
```

Comment:
Should be `RunResponse` and we should change the return type to be `Promise<RunResponse>`